### ISSUE TITLE
Issue with Lazy Loading Routes in Standalone API

### ISSUE THREAD

**Issue by thoussem** [angular/angular#56865]
### Which @angular/* package(s) are the source of the bug?

router

### Is this a regression?

Yes

### Description

Here's the corrected bug report for the Angular team:

---

**Title: Issue with Lazy Loading Routes in Standalone API**

**Description:**

After upgrading to the Angular standalone API and removing the NgModules, I encountered an issue with lazy loading routes. Below are the details of the problem:

**Before upgrading to the Angular standalone API:**

```typescript
import { Routes } from '@angular/router';

import { ScreenDialogComponent } from '@shared/components/screen-dialog/screen-dialog.component';
import { AirportAirplaneeditComponent } from './airplaneedit/airport-airplaneedit.component';
import { AirportAirplanesComponent } from './airplanes/airport-airplanes.component';
import { AirportEdit0Component } from './edit-0/airport-edit-0.component';
import { AirportEditCollection0Component } from './edit-collection-0/airport-edit-collection-0.component';
import { AirportView0Component } from './view-0/airport-view-0.component';

/**
 * Defines the routes for the Airport feature.
 */
const routes: Routes = [
  {
    path: 'airplaneedit/:id',
    component: AirportAirplaneeditComponent,
  },
  {
    path: 'airplaneedit',
    component: AirportAirplaneeditComponent,
  },
  {
    path: 'view-0/:id',
    component: AirportView0Component,
  },
  {
    path: 'view-0',
    component: AirportView0Component,
  },
  {
    path: 'airplanes',
    component: AirportAirplanesComponent,
    children: [
      {
        path: 'airport',
        component: ScreenDialogComponent,
        outlet: 'dialog',
        // Inside the dialog outlet, I can open any screen of the same module
        children: [
          {
            path: '',
            loadChildren: () =>
              import('@features/airport/airport.module').then((m) => m.AirportModule),
          },
        ],
      },
      {
        path: 'form',
        component: ScreenDialogComponent,
        outlet: 'dialog',
        children: [
          {
            path: '',
            loadChildren: () => import('@features/form/form.module').then((m) => m.FormModule),
          },
        ],
      },
    ],
  },
];

/**
 * The Airport routing module
 *
 * @stable
 */
@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class AirportRoutingModule {}
```

I was able to load any component of the Airport module inside the **dialog** outlet using `loadChildren` and lazy load the same module.

**After upgrading to the new standalone API and removing the NgModules:**

```typescript
import { Routes } from '@angular/router';

import { ScreenDialogComponent } from '@shared/components/screen-dialog/screen-dialog.component';
import { AirportAirplaneeditComponent } from './airplaneedit/airport-airplaneedit.component';
import { AirportAirplanesComponent } from './airplanes/airport-airplanes.component';
import { AirportEdit0Component } from './edit-0/airport-edit-0.component';
import { AirportEditCollection0Component } from './edit-collection-0/airport-edit-collection-0.component';
import { AirportView0Component } from './view-0/airport-view-0.component';

/**
 * Defines the routes for the Airport feature.
 */
export const routes: Routes = [
  {
    path: 'airplaneedit/:id',
    component: AirportAirplaneeditComponent,
  },
  {
    path: 'airplaneedit',
    component: AirportAirplaneeditComponent,
  },
  {
    path: 'view-0/:id',
    component: AirportView0Component,
  },
  {
    path: 'view-0',
    component: AirportView0Component,
  },
  {
    path: 'airplanes',
    component: AirportAirplanesComponent,
    children: [
      {
        path: 'airport',
        component: ScreenDialogComponent,
        outlet: 'dialog',
        // Inside the dialog outlet, I can't implement the recursivity
        children: [
          {
            path: '',
            loadChildren: () =>
              import('@features/airport/airport.module').then((m) => m.routes),
          },
        ],
      },
      {
        path: 'form',
        component: ScreenDialogComponent,
        outlet: 'dialog',
        children: [
          {
            path: '',
            loadChildren: () => import('@features/form/form.module').then((m) => m.routes),
          },
        ],
      },
    ],
  },
];
```

I encountered a runtime error when attempting to open the dialog.

**Error:**
```
core.mjs:6620 ERROR ReferenceError: require_aireport_routing_module is not defined
    at aireport-routing.module.ts:102:15
    at _ZoneDelegate.invoke (zone.js:365:28)
    at Object.onInvoke (core.mjs:15749:33)
    at _ZoneDelegate.invoke (zone.js:364:34)
    at _ZoneImpl.run (zone.js:111:43)
    at zone.js:2438:40
    at _ZoneDelegate.invokeTask (zone.js:400:33)
    at core.mjs:15434:55
    at AsyncStackTaggingZoneSpec.onInvokeTask (core.mjs:15434:36)
    at _ZoneDelegate.invokeTask (zone.js:399:38)

```
**Reproduction Steps:**
1. Upgrade to Angular standalone API.
2. Remove NgModules.
3. Implement lazy loading as shown in the example above.

**Expected Behavior:**
The application should successfully lazy load the routes without any errors.

**Environment:**
- Angular Version: 18
- Node.js Version: 18.20.3

Any suggestions on how to fix this would be greatly appreciated.

Thank you!

**Comment by atscott** [comment]
Duplicate of #5489. See comment here for an explanation: https://github.com/angular/angular/issues/54898#issuecomment-2004029179

TL;DR - just remove the lazy loading. It’s not providing any benefit since you’ve already loaded the file.
